conf/root.go

```diff
@@ -101,5 +134,77 @@ var (
 func init() {
 	createContextCmd.Flags().StringVarP(&confFileTemplate, "templatefile", "f", "", "-f <config file to build context from>")
 	dumpCmd.Flags().StringVarP(&dumpOutfile, "outfile", "o", "", "-o <outfile")
-	RootCmd.AddCommand(dumpCmd, createContextCmd, getContextCmd, getContextsCmd, useContextCmd, deleteContextCmd)
+	RootCmd.AddCommand(dumpCmd, createContextCmd, currentContextCmd, getContextsCmd, useContextCmd, deleteContextCmd, setCmd, editCmd)
+}
+
+func setConfigWithViper(key string, value string) error {
+	v := viper.New()
+	cname, err := GetCurrentContextName()
+	if err != nil {
+		return err
+	}
+	basedir, err := getContextBaseDir()
+	if err != nil {
+		return err
+	}
+	v.SetConfigFile(filepath.Join(basedir, cname))
+	v.SetConfigType("toml")
+	if err = v.ReadInConfig(); err != nil {
+		return err
+	}
+	if !v.IsSet(key) {
+		return fmt.Errorf("Key '%s' not found in config.", key)
+	}
+	v.Set(key, value)
+	var config ContextConfig
+	if err = v.Unmarshal(&config); err != nil {
+		return err
+	}
+	if err = writeConfigToCurrentContext(&config); err != nil {
+		return err
+	}
+	return nil
+}
+
+func editConfig(contextName string) error {
+	tempFile, err := os.CreateTemp("", contextName)
+	if err != nil {
+		return err
+	}
+	tempFile.Close()
+	defer os.Remove(tempFile.Name())
+
+	var existingConfig ContextConfig
+	if err = readConfigFromContext(contextName, &existingConfig); err != nil {
+		return err
+	}
+
+	if err = writeConfigToFile(tempFile.Name(), &existingConfig); err != nil {
+		return err
+	}
+
+	editor := os.Getenv("EDITOR")
+	if editor == "" {
+		return errors.New("Please set the $EDITOR environment variable to your preferred text editor.")
+	}
```

Comment:
I don't seem to have this env var set. Would it be better to store this in the ExecutionConfig?

Reply:
$EDITOR is pretty standard on unix systems, but I'm also open to setting it in the ExecutionConfig (I believe git also has a config setting for editor)